hoomd3/classic/butane/metad/butane_Metadynamics.py

```diff
@@ -0,0 +1,317 @@
+    dihedral.params["HCCH"] = dict(k1=0.0, k2=0.0, k3=0.3, k4=0.0)
+
+    lj_special_pairs = hoomd.md.special_pair.LJ()
+    lj_special_pairs.params["CCCC"] = dict(epsilon=0.07, sigma=3.55)
+    lj_special_pairs.params["HCCH"] = dict(epsilon=0.03, sigma=2.42)
+    lj_special_pairs.params["HCCC"] = dict(epsilon=sqrt(0.07 * 0.03), sigma=sqrt(3.55 * 2.42))
+    lj_special_pairs.r_cut["CCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCC"] = 12.0
+    lj_special_pairs.r_cut["HCCH"] = 12.0
+
+    coulomb_special_pairs = hoomd.md.special_pair.Coulomb()
+    coulomb_special_pairs.params["CCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCC"] = dict(alpha=0.5)
+    coulomb_special_pairs.params["HCCH"] = dict(alpha=0.5)
+    coulomb_special_pairs.r_cut["HCCH"] = 12.0
+    coulomb_special_pairs.r_cut["CCCC"] = 12.0
+    coulomb_special_pairs.r_cut["HCCC"] = 12.0
+    integrator = hoomd.md.Integrator(dt=dt)
+    integrator.forces.append(lj)
+    integrator.forces.append(coulomb[0])
+    integrator.forces.append(coulomb[1])
+    integrator.forces.append(harmonic)
+    integrator.forces.append(angle)
+    integrator.forces.append(dihedral)
+    integrator.forces.append(lj_special_pairs)
+    integrator.forces.append(coulomb_special_pairs)
+    nvt = hoomd.md.methods.Langevin(filter=hoomd.filter.All(), kT=kT)
+    integrator.methods.append(nvt)
+    sim.operations.integrator = integrator
+
+    return sim
+
+
+# %%
+def get_args(argv):
+    available_args = [
+        ("well-tempered", "w", bool, 0, "Whether to use well-tempered metadynamics"),
+        ("use-grids", "g", bool, 0, "Whether to use grid acceleration"),
+        ("log", "l", bool, 0, "Whether to use a callback to log data into a file"),
+        ("time-steps", "t", int, 5e5, "Number of simulation steps"),
+    ]
+
+    parser = argparse.ArgumentParser(description="Example script to run metadynamics")
+    for name, short, T, val, doc in available_args:
+        parser.add_argument("--" + name, "-" + short, type=T, default=T(val), help=doc)
+
+    return parser.parse_args(argv)
+
+
+# %%
+def main(argv=[]):
+    args = get_args(argv)
+
+    cvs = [DihedralAngle([0, 4, 7, 10])]
+
+    height = 0.01  # kJ/mol
+    sigma = 0.1  # radians
+    deltaT = 5000 if args.well_tempered else None
+    stride = 50  # frequency for depositing gaussians
+    timesteps = args.time_steps
+    ngauss = timesteps // stride  # total number of gaussians
+
+    grid = Grid(lower=(-pi,), upper=(pi,), shape=(64,), periodic=True)
+    grid = grid if args.use_grids else None
+
+    method = Metadynamics(cvs, height, sigma, stride, ngauss, deltaT=deltaT, kB=kB, grid=grid)
+
+    hills_file = "hills.dat"
+    callback = MetaDLogger(hills_file, stride) if args.log else None
+
+    tic = time.perf_counter()
+    pysages.run(method, generate_context, timesteps, callback)
+    toc = time.perf_counter()
+    print(f"Completed the simulation in {toc - tic:0.4f} seconds.")
```

Comment:
Also this script is not printing out results (only `hills.dat`, which is only part of the results), see the comments above, incl. pickling https://github.com/SSAGESLabs/PySAGES-examples/pull/14/files

Reply:
Plot pand pickle file in https://github.com/SSAGESLabs/PySAGES-examples/pull/15/commits/6da8457e8ad8671a45b0d8e0a078158159b39e9e